rcl/test/rcl/test_arguments.cpp

```diff
@@ -751,12 +808,33 @@ TEST_F(CLASSNAME(TestArgumentsFixture, RMW_IMPLEMENTATION), test_param_argument_
   char ** parameter_files = NULL;
   ret = rcl_arguments_get_param_files(&parsed_args, alloc, &parameter_files);
   ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-  EXPECT_STREQ("parameter_filepath1", parameter_files[0]);
-  EXPECT_STREQ("parameter_filepath2", parameter_files[1]);
+  EXPECT_STREQ(parameters_filepath1.c_str(), parameter_files[0]);
+  EXPECT_STREQ(parameters_filepath2.c_str(), parameter_files[1]);
   for (int i = 0; i < parameter_filecount; ++i) {
     alloc.deallocate(parameter_files[i], alloc.state);
   }
+
   alloc.deallocate(parameter_files, alloc.state);
+
+  rcl_params_t * params = NULL;
+  ret = rcl_arguments_get_param_overrides(&parsed_args, &params);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+    rcl_yaml_node_struct_fini(params);
+  });
+  EXPECT_EQ(2U, params->num_nodes);
+
+  rcl_variant_t * param_value =
+    rcl_yaml_node_struct_get("some_node", "int_param", params);
+  ASSERT_TRUE(NULL != param_value);
+  ASSERT_TRUE(NULL != param_value->integer_value);
+  EXPECT_EQ(1, *(param_value->integer_value));
+
+  param_value = rcl_yaml_node_struct_get("another_node", "double_param", params);
+  ASSERT_TRUE(NULL != param_value);
+  ASSERT_TRUE(NULL != param_value->double_value);
+  EXPECT_DOUBLE_EQ(1.0, *(param_value->double_value));
+
   EXPECT_EQ(RCL_RET_OK, rcl_arguments_fini(&parsed_args));
```

Comment:
Finalize call should be done in a `OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT`.

Reply:
Done in https://github.com/ros2/rcl/pull/508/commits/c1f4c3ce1dc69cf0d29cf1c72873d582efa7a557